### PR TITLE
Feat: Buyer should be able to list products from vendors

### PR DESCRIPTION
**Pull request by niyibi250**
### What does this PR do?

This pull request implements the AvailableProducts function, which retrieves available products from the database with pagination support. The function allows clients to request available products with optional pagination parameters (limit and page). If pagination parameters are not provided, default values are used (limit=10, page=1).

### Description

- [ ]  **Implementing the `/getAvailableProducts` route**

This function should execute a database query to fetch available products where the `isAvailablefield` is set to` true`.
Pagination should be applied to limit the number of products returned per page and to skip the appropriate number of records based on the page number.
The function should include the category relation to eagerly load associated categories for each product.
Handling pagination:

If pagination parameters (`limit and page`) are not provided in the query, default values should be used (`limit=10, page=1`).

- [ ]  **Constructing a JSON response**

Once the products are fetched, the function should construct a JSON response containing the retrieved products, total pages, and the current page.

Total pages should be calculated based on the total count of available products and the specified limit.

- [ ]  **Implementing error handling**

The function should include appropriate error handling to handle any errors that may occur during the execution of the function (e.g., database query error, server error).
If an error occurs, the function should return a 500-status code along with an error message.

 
- [ ] **Testing**

Test the `getAvailableProducts` route to ensure it retrieves available products correctly with the expected pagination behavior.
Verify that error handling is working as expected and that appropriate responses are returned in case of errors.

Testing: I have tested the getAvailableProducts route locally and verified that it retrieves available products correctly with the expected pagination behavior.

### Related Issues:

#37

### Screenshot:

Swagger:

image

Testing:

test shot

### CODE REVIEW

File: src/controller/list_products_from_vendorsController.ts
Comment (wayneleon1) [inline src/controller/list_products_from_vendorsController.ts:1]: You don't need to retrieve all the information of vendor , try to select only firstName or lastName

File: src/routes/list_products_from_vendorsRoute.ts
Comment (wayneleon1) [inline src/routes/list_products_from_vendorsRoute.ts:1]: and also i think this file have too long name : list_products_from_vendorsController 
try to use readable  file name  which is not too long

**Review by wayneleon1 — CHANGES_REQUESTED** [review]

File: src/controller/list_products_from_vendorsController.ts
Comment (niyibi250) [reply src/controller/list_products_from_vendorsController.ts:1]: ok finished now you can review again

**Review by bertrandshema — CHANGES_REQUESTED** [review]
Since we have a productController.ts, product.test.ts, and product.routes.ts we do need to create other files. your issues is related to products can you please use those files no need to create other files.

Also all other product endpoints are using a **'/api/v1/product/....'** lets please use the same format for this one

**Comment by ambroisegithub** [comment]
And try to test all functions you see the test coverage was reduced -0.1%....

**Comment by ambroisegithub** [comment]
And  combine your multiple commits into one.

**Pull request closed without merging**